### ISSUE TITLE
Azure SQL transient error

### ISSUE THREAD

**Issue by conreaux** [microsoft/durabletask-mssql#182]
We occasionally encounter a transient Azure SQL error 42102 (see definition below) in our DF application. Our IT department is investigating root cause. It can hit any DF in our app. Is this a candidate to add to the [transient errors list](https://github.com/microsoft/durabletask-mssql/blob/225738761880c13c82348f9d9b61a17dcbd14290/src/DurableTask.SqlServer/SqlUtils.cs#L532C11-L532C11) for automatic retry?

We could add custom retry handlers to activity functions, but that doesn't address the other cases. Other suggestions for handling such an error?

_Reason: Unable to set Deny Public Network Access to Yes since there is no private endpoint enabled to access the server. Please set up private endpoints and retry the operation ([https://learn.microsoft.com/azure/sql-database/sql-database-private-endpoint-overview#how-to-set-up-private-link-for-azure-sql-database](https://learn.microsoft.com/en-us/azure/sql-database/sql-database-private-endpoint-overview#how-to-set-up-private-link-for-azure-sql-database))._

**Comment by cgillum** [comment]
@conreaux the description of 47073 sounds like it's not supposed to be transient, but you're observing it happening transiently? Do I understand that correctly?

**Comment by conreaux** [comment]
Yes, I agree that it should not be transient. However, we encounter this error intermittently and typically for a duration of 3-5 minutes. Our IT and Microsoft support are investigating, but it does seem related to DNS resolution somehow defaulting to the public endpoint even though a private endpoint is configured on the Azure SQL database (our app is running in an isolated App Service Environment). Similar issues are mentioned in this [Reddit post](https://www.reddit.com/r/AZURE/comments/ys8nxs/dns_keeps_swapping_between_public_and_private/).  I understand that it probably does not fit into the transient errors list and probably not much that can be done in the MSSQL provider, but it does cause problems since any DF call can fail. Maybe just looking for error handling advice to make our app fault tolerant in cases where database errors may escape the retry logic.

**Comment by cgillum** [comment]
I think your suggestion of allowing custom status codes to be configured makes sense for a case like this. I’m supportive of this idea.
